node/src/blockchain/blockchain_bridge.rs

```diff
@@ -1648,25 +1649,123 @@ mod tests {
+            (
+                TestServer::start(
+                    port,
+                    vec![br#"{"jsonrpc":"2.0","id":0,"result":someGarbage}"#.to_vec()],
+                ),
+                server_url,
+            )
         }
+        fn assert_on_persistent_config_connection(
+            blockchain_bridge_addr_rv: Receiver<Addr<BlockchainBridge>>,
+            wallet: Wallet,
+            expected_gas_price: u64,
+        ) {
+            let blockchain_bridge_addr = blockchain_bridge_addr_rv.try_recv().unwrap();
+            let msg = AssertionsMessage {
+                assertions: Box::new(move |bb: &mut BlockchainBridge| {
+                    // the assertion on this call is made outside and checks
+                    // the received request
+                    let _result = bb.blockchain_interface.helpers().get_masq_balance(&wallet);
+
+                    assert_eq!(
+                        bb.persistent_config.gas_price().unwrap(),
+                        expected_gas_price
+                    );
+
+                    System::current().stop();
+                }),
+            };
+            //this assertion message will eventually stop the system
+            blockchain_bridge_addr.try_send(msg).unwrap();
+        }
+        fn assert_on_blockchain_interface_connection(test_server: &TestServer, wallet: Wallet) {
+            let requests = test_server.requests_so_far();
+            let bodies: Vec<Value> = requests
+                .into_iter()
+                .map(|request| serde_json::from_slice(&request.body()).unwrap())
+                .collect();
+            assert_eq!(
+                bodies[0]["params"][0]["data"].to_string()[35..75],
+                wallet.to_string()[2..]
+            );
+            assert_eq!(
+                bodies[0]["params"][0]["to"],
+                format!("{:?}", TEST_DEFAULT_CHAIN.rec().contract)
+            );
+        }
+
+        let data_dir = ensure_node_home_directory_exists(test_module, test_name);
+        let gas_price = 444;
+        arrange_db(&data_dir, gas_price);
+        let (test_server, server_url) = launch_prepared_test_server();
+        let wallet = make_wallet("abc");
+        let mut bootstrapper_config = BootstrapperConfig::new();
+        bootstrapper_config
+            .blockchain_bridge_config
+            .blockchain_service_url_opt = Some(server_url);
+        bootstrapper_config.blockchain_bridge_config.chain = TEST_DEFAULT_CHAIN;
+        bootstrapper_config.data_directory = data_dir;
+        let (tx, blockchain_bridge_addr_rv) = bounded(1);
+        let address_leaker = SubsFactoryTestAddrLeaker { address_leaker: tx };
+        let system = System::new(test_name);
+
+        act(bootstrapper_config, address_leaker);
+
+        assert_on_persistent_config_connection(
+            blockchain_bridge_addr_rv,
+            wallet.clone(),
+            gas_price,
+        );
+        assert_eq!(system.run(), 0);
```

Comment:
If you moved the stop-the-system call to just before this line, I think it would work, and if it did, all the System-related stuff would be in the same place.